src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1989,7 +1943,7 @@ internal double GetPortVerticalOffset(PortModel portModel)
         ///     Reads inputs list and adds ports for each input.
         /// </summary>
         [Obsolete("RegisterInputPorts is deprecated, please use the InPortNamesAttribute, InPortDescriptionsAttribute, and InPortTypesAttribute instead.")]
-        public void RegisterInputPorts(IEnumerable<PortData> portDatas)
+        internal void RegisterInputPorts(IEnumerable<PortData> portDatas)
```

Comment:
@sm6srw I feel these two apis (register) might need to be better evaluated for removal, I am not sure we have provided the same level of functionality with the attributes as these methods provide, especially for dynamic nodes. Maybe file a followup to evaluate them and add a TODO?